Restoring optional tools fails

```msbuild
C:\git\corefx>.\sync.cmd
Tools are already initialized.
Running: C:\git\corefx\Tools\dotnetcli\dotnet msbuild /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false /l:BinClashLogger,Tools\Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log  /flp:v=normal;LogFile=sync.log /t:Sync
C:\git\corefx\external\test-runtime\optional.csproj(2,3): warning MSB4011: "C:\git\corefx\dependencies.props" cannot be imported again. It was already imported at "C:\git\corefx\Directory.Build.props (79,3)". This is most likely a build authoring error. This subsequent import will be ignored. [C:\git\corefx\build.proj]
C:\git\corefx\external\test-runtime\optional.csproj(2,3): warning MSB4011: "C:\git\corefx\dependencies.props" cannot be imported again. It was already imported at "C:\git\corefx\Directory.Build.props (79,3)". This is most likely a build authoring error. This subsequent import will be ignored. [C:\git\corefx\build.proj]
    Restoring packages for C:\git\corefx\external\test-runtime\optional.csproj...
C:\git\corefx\external\test-runtime\optional.csproj : error NU1101: Unable to find package TestILC.x86ret. No packages exist with this id in source(s): https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json, https://dotnet.myget.org/F/dotnet-core/api/v3/index.json, https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json, nuget.org [C:\git\corefx\build.proj]
C:\git\corefx\external\test-runtime\optional.csproj : error NU1101: Unable to find package Microsoft.DotNet.IBCMerge. No packages exist with this id in source(s): https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json, https://dotnet.myget.org/F/dotnet-core/api/v3/index.json, https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json, nuget.org [C:\git\corefx\build.proj]
C:\git\corefx\external\test-runtime\optional.csproj : error NU1101: Unable to find package TestILC.amd64ret. No packages exist with this id in source(s): https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json, https://dotnet.myget.org/F/dotnet-core/api/v3/index.json, https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json, nuget.org [C:\git\corefx\build.proj]
C:\git\corefx\external\test-runtime\optional.csproj : error NU1101: Unable to find package TestILC.armret. No packages exist with this id in source(s): https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json, https://dotnet.myget.org/F/dotnet-core/api/v3/index.json, https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json, nuget.org [C:\git\corefx\build.proj]
    Restore failed in 1.14 sec for C:\git\corefx\external\test-runtime\optional.csproj.
C:\git\corefx\Tools\OptionalTooling.targets(84,5): error MSB3073: The command "C:\git\corefx\Tools/dotnetcli/dotnet.exe restore C:\git\corefx\\external\test-runtime\optional.csproj --configfile C:\git\corefx\bin/obj/generatedNuGetConfig/optional.NuGet.Config /p:RestoreOutputPath=C:\git\corefx\Tools/optional-tool-runtime\" exited with code 1. [C:\git\corefx\build.proj]

Build FAILED.
```

cc @eerhardt @weshaggard

--- a/external/test-runtime/optional.csproj
+++ b/external/test-runtime/optional.csproj
@@ -1,6 +1,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>net45</TargetFramework>
+    <!-- $(RestoreSources) is set in the root Directory.Build.props and will override any sources specified
+         in the generated NuGet.config file. Blank it out here so the NuGet.config sources are used.
+     -->
+    <RestoreSources></RestoreSources>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.DotNet.IBCMerge" Version="$(MicrosoftDotNetIBCMergePackageVersion)" />

